### PR TITLE
Basic Wizards

### PR DESCRIPTION
**Pull request by lizardqueenlexi**
## About The Pull Request

Makes NPC wizard mobs into basic mobs. These aren't actually used anywhere, since their away mission was removed, so I figured I'd have some fun with them.

image

Noteworthy changes are as follows:
- Wizard mobs can now wear any of the four basic colors of wizard robe and hat. Rarely, they will wear the witch ("Marisa") outfit instead. There was going to be a rare Tape Wizard spawn too, but the outfit doesn't work correctly for some reason (see comments below).
- Wizard mobs no longer have a set spell loadout. Instead, they always receive a random **targeted, primary spell**, a random **untargeted, secondary spell**, and the spell **Blink**.
- Wizard subtypes (or var-edited wizards) can have any of the above specified rather than random.
- Wizard AI will try to avoid ever being in melee, and will fire off spells whenever possible with the priority order of primary > secondary > blink. There is a mandatory 1-second waiting period between casts.
- Wizard mobs use "lesser" versions of Fireball and Blink. Lesser Fireball does a little less damage and has a smaller explosion, though it is still extremely dangerous. Lesser Blink simply has a smaller blink radius so that wizard mobs don't just decide to leave.

Depending on their spell loadout, wizards can be _incredibly_ dangerous mobs - stunning you with tesla blasts, shooting you with fireballs, and more. Even weakened, fireball is the nastiest of these by far, able to set you on fire or remove your limbs if you aren't properly protected. Watch out!

The random lists have been kept pretty small, since I only wanted to use spells that I know for sure work. Spell cards is pretty weak in AI hands, since they don't take advantage of the fact that a human player can spam it, but I wanted there to be a little variety in primary attacks.

I included an UpdatePaths script here in case a downstream is using these, but I doubt it was necessary.
## Why It's Good For The Game

Removes another simplemob.

The actual impact of this change is negligible, since this is an unused mob, but it's possible that this lays the groundwork for these actually being used - particularly since mappers can make them as powerful or weak as they want by specifying a spell loadout. Wizards may make for a tough boss challenge for a ruin or away mission, or a very mean surprise for an admin to drop on an unsuspecting station.
## Changelog
As this is an unused mob, there are no player-facing changes.

### CODE REVIEW

**Comment by Rex9001** [comment]
I might be wrong and thinking of something else, but arent these used in a lavaland ruin?

**Comment by lizardqueenlexi** [comment]
> I might be wrong and thinking of something else, but arent these used in a lavaland ruin?

You're thinking of "dark wizards", which are completely unrelated despite the name.

File: code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		/obj/effect/mob_spawn/corpse/human/wizard,
+		/obj/effect/mob_spawn/corpse/human/wizard/red,
+		/obj/effect/mob_spawn/corpse/human/wizard/yellow,
+		/obj/effect/mob_spawn/corpse/human/wizard/black,
+	)
+	/// A specified wizard corpse spawner to use. If null, picks from the list above instead.
+	var/selected_outfit
+
+	/// The wizard's "main" attack, a targeted spell like fireball
+	var/datum/action/cooldown/spell/targeted_spell
```
Comment (Jacquerel) [inline code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm:36]: What are we holding this and `secondary_spell` in memory for? Doesn't look necessary.

File: code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	ai_controller.set_blackboard_key(BB_WIZARD_TARGETED_SPELL, targeted_spell)
+
+	secondary_spell = new secondary_spell_path(src)
+	secondary_spell.spell_requirements &= ~(SPELL_REQUIRES_HUMAN|SPELL_REQUIRES_WIZARD_GARB|SPELL_REQUIRES_MIND)
+	secondary_spell.Grant(src)
+	ai_controller.set_blackboard_key(BB_WIZARD_SECONDARY_SPELL, secondary_spell)
+
+	blink_spell = new(src)
+	blink_spell.spell_requirements &= ~(SPELL_REQUIRES_HUMAN|SPELL_REQUIRES_WIZARD_GARB|SPELL_REQUIRES_MIND)
+	blink_spell.outer_tele_radius = 3
```
Comment (Jacquerel) [inline code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm:90]: At this point maybe just make a subtype?

File: code/modules/mob_spawn/corpses/mob_corpses.dm [hunk 3]
```diff
@@ -221,13 +221,34 @@
 	facial_haircolor = COLOR_WHITE
 	skin_tone = "caucasian1"
 
+/obj/effect/mob_spawn/corpse/human/wizard/red
+	outfit = /datum/outfit/wizardcorpse/red
+
+/obj/effect/mob_spawn/corpse/human/wizard/yellow
+	outfit = /datum/outfit/wizardcorpse/yellow
+
+/obj/effect/mob_spawn/corpse/human/wizard/black
```
Comment (Jacquerel) [inline code/modules/mob_spawn/corpses/mob_corpses.dm:230]: Where's the tape wizard?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by lizardqueenlexi** [comment]
For reasons unbeknownst to science, the tape wizard hat looks like this when rendered with `apply_dynamic_human_appearance`:

image

So tragically, I have _not_ included it in the outfit list right now. I left a comment about it in the list in case anyone does manage to fix this. On the bright side, I did also make a corpse spawner for the Marisa Wizard, and gave basic wizard mobs a rare chance of spawning in that outfit.

File: code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	/// A specified wizard corpse spawner to use. If null, picks from the list above instead.
+	var/selected_outfit
+
+	/// Typepath for the wizard's targeted spell. If null, selects randomly.
+	var/targeted_spell_path
+	/// List of possible targeted spells to pick from
+	var/static/list/targeted_spell_list = list(
+		/datum/action/cooldown/spell/pointed/projectile/fireball/lesser,
+		/datum/action/cooldown/spell/pointed/projectile/lightningbolt,
+		/datum/action/cooldown/spell/pointed/projectile/spell_cards,
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm:43]: Spell Cards is probably no good considering it's not just fire and forget and is charged based when the others is not

File: code/modules/mob_spawn/corpses/mob_corpses.dm [hunk 5]
```diff
@@ -221,13 +221,48 @@
...
+
+/datum/outfit/wizardcorpse/yellow
+	suit = /obj/item/clothing/suit/wizrobe/yellow
+	head = /obj/item/clothing/head/wizard/yellow
+
+/datum/outfit/wizardcorpse/black
+	suit = /obj/item/clothing/suit/wizrobe/black
+	head = /obj/item/clothing/head/wizard/black
+
+/datum/outfit/wizardcorpse/marisa
```
Comment (Fikou) [inline code/modules/mob_spawn/corpses/mob_corpses.dm:258]: should probably use her unique shoes

File: code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm (hunk 4 above)
Comment (lizardqueenlexi) [reply code/modules/mob/living/basic/ruin_defender/wizard/wizard.dm]: That's a good point. I don't really like having only two main spells, but spell cards are really, really weak for AI to use. I've removed it.

**Pull request merged**